src/Native/System.Native/pal_networking.cpp

```diff
@@ -703,3 +708,119 @@ extern "C" int32_t TryGetIPPacketInformation(uint8_t* messageHeader, int32_t isI
 
     return 0;
 }
+
+static bool GetMulticastOptionName(int32_t multicastOption, bool isIPv6, int& optionName)
+{
+    switch (multicastOption)
+    {
+        case PAL_MULTICAST_ADD:
+            optionName = isIPv6 ? IPV6_ADD_MEMBERSHIP : IP_ADD_MEMBERSHIP;
+            return true;
+
+        case PAL_MULTICAST_DROP:
+            optionName = isIPv6 ? IPV6_DROP_MEMBERSHIP : IP_DROP_MEMBERSHIP;
+            return true;
+
+        default:
+            return false;
+    }
+}
+
+extern "C" int32_t GetIPv4MulticastOption(int32_t socket, int32_t multicastOption, IPv4MulticastOption* option)
+{
+    if (option == nullptr)
```

Comment:
No: this is effectively public surface area, so we should be coding defensively against upstream bugs. There are other portions of this file that need to be updated to do this as well.